modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/DistributionZoneCausalityDataNodesTest.java

```diff
@@ -1013,62 +996,128 @@ void simpleTopologyChanges() throws Exception {
 
-        Set<String> expectedDefaultZoneDataNodes = Set.of(NODE_0.name(), NODE_2.name());
-        Set<String> expectedZone1DataNodes = Set.of(NODE_0.name(), NODE_1.name(), NODE_2.name());
-        Set<String> expectedZone2DataNodes = Set.of(NODE_0.name());
-        Set<String> expectedZone3DataNodes = Set.of(NODE_0.name(), NODE_1.name());
+        latch.await();
+
+        // Check that data nodes values are idempotented.
+        // Data nodes from the meta storage manager only used to calculate current data nodes because the meta storage are updated and
+        // topology augmentation maps were cleared.
+        checkDataNodes(
+                topologyUpdateRevision.get(),
+                expectedDefaultZoneDataNodes,
+                expectedZone1DataNodes,
+                expectedZone2DataNodes,
+                expectedZone3DataNodes
+        );
 
-        // Check that data nodes values are changed according to scale up and down timers.
-        // Data nodes from the meta storage manager and topology augmentation maps are used to calculate current data nodes.
         checkDataNodes(
-                topologyRevision2,
+                metaStorageManager.appliedRevision(),
                 expectedDefaultZoneDataNodes,
                 expectedZone1DataNodes,
                 expectedZone2DataNodes,
                 expectedZone3DataNodes
         );
+    }
+
+    /**
+     * Check that data nodes calculation is idempotented.
+     * The current data nodes value which was calculated by data nodes from the meta storage manager and topology augmentation maps equals
+     * to the data nodes value which were calculated by data nodes from the meta storage manager after topology augmentation maps
+     * were cleared.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    void checkDataNodesRepeatedOnNodeRemoved() throws Exception {
+        prepareZonesWithTwoDataNodes();
+
+        prepareZonesTimerValuesToTest();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        AtomicBoolean reached = new AtomicBoolean();
 
-        // Check that data nodes values are not changed in the meta storage.
-        checkThatDataNodesIsNotChangedInMetastorage();
+        Set<String> expectedDefaultZoneDataNodes = Set.of(NODE_0.name());
+        Set<String> expectedZone1DataNodes = Set.of(NODE_0.name(), NODE_1.name());
+        Set<String> expectedZone2DataNodes = Set.of(NODE_0.name());
+        Set<String> expectedZone3DataNodes = Set.of(NODE_0.name(), NODE_1.name());
 
-        // Unblock scale up and down tasks.
-        scaleUpLatch.countDown();
-        scaleDownLatch.countDown();
+        AtomicLong topologyUpdateRevision = new AtomicLong();
 
-        // Check that data nodes values are changed in the meta storage after scale up and scale down tasks updated the meta storage.
-        checkThatDataNodesIsChangedInMetastorage(
+        WatchListener testListener = createLogicalTopologyWatchListenerToCheckDataNodes(
```

Comment:
`checkThatDataNodesIsChangedInMetastorage` is not used anymore, looks strange

Reply:
Fixed. Now it is used.